modules/javafx.graphics/src/test/java/test/javafx/scene/layout/SnappingTest.java

```diff
@@ -0,0 +1,165 @@
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.javafx.scene.layout;
+
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class SnappingTest {
+
+    private Stage stage;
+
+    @AfterEach
+    void tearDown() {
+        if (stage != null) {
+            stage.hide();
+            stage = null;
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = { "getContainerInstruction" })
+    void testContainerSnappingScale100(ContainerInstruction<Region> containerInstruction) {
+        testContainerSnappingImpl(containerInstruction, 1);
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = { "getContainerInstruction" })
+    void testContainerSnappingScale125(ContainerInstruction<Region> containerInstruction) {
+        testContainerSnappingImpl(containerInstruction, 1.25);
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = { "getContainerInstruction" })
+    void testContainerSnappingScale150(ContainerInstruction<Region> containerInstruction) {
+        testContainerSnappingImpl(containerInstruction, 1.5);
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = { "getContainerInstruction" })
+    void testContainerSnappingScale175(ContainerInstruction<Region> containerInstruction) {
+        testContainerSnappingImpl(containerInstruction, 1.75);
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = { "getContainerInstruction" })
+    void testContainerSnappingScale200(ContainerInstruction<Region> containerInstruction) {
+        testContainerSnappingImpl(containerInstruction, 2);
```

Comment:
minor: do you think there is a need for other scale factors?

Reply:
I don't think so. We could add more but I don't see how this will give us any more information/benefits.